src/pc/sim_pc.c

```diff
@@ -107,14 +107,20 @@ void func_800E7D08(void) {
 }
 
 void LoadStageTileset(u8* pTilesetData, size_t len, s32 y) {
+    const u32 MaxBlockCount = 0x20;
     RECT rect;
     u8* pTilesetDataSrc;
     u32 i;
     u32 blockCount;
 
     rect.w = 0x20;
     rect.h = 0x80;
-    blockCount = CLAMP_MAX(len >> 13, 0x20);
+    blockCount = len >> 13;
+    if (blockCount > MaxBlockCount) {
+        WARNF("tileset too large (%d > %d): it wil not be fully read",
```

Comment:
I thought the issue was that the tileset is too small? What about the case where it's wanting to load more bytes than the file has?

Reply:
I guess I'm not understanding what is going on. It seems to me there can be two cases:

The file is smaller than the number of bytes the game wants to load: we need to only load what actually exists to avoid a buffer overrun. We should warn about this since it's difference in behavior vs. psx, where I guess random junk will be written at the end?

The file is larger than the number of bytes the game wants to load: this shouldn't cause a crash but makes sense to warn about too.

It seems like we are warning about the latter case and not the former?

Reply:
Ok whichever you think makes the most sense is fine with me